vaultpress.php

```diff
@@ -9,18 +9,69 @@
  * License: GPL2+
  * Text Domain: vaultpress
  * Domain Path: /languages/
+ *
+ * @package VaultPress
+ */
+
+// don't call the file directly.
+defined( 'ABSPATH' ) || die();
+
+define( 'VAULTPRESS__MINIMUM_PHP_VERSION', '5.3.2' );
+define( 'VAULTPRESS__VERSION', '2.0-beta' );
+define( 'VAULTPRESS__PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+
+/**
+ * First, we check for our supported version of PHP. If it fails,
+ * we "pause" VaultPress by ending the loading process and displaying an admin_notice to inform the site owner.
  */
+if ( version_compare( phpversion(), VAULTPRESS__MINIMUM_PHP_VERSION, '<' ) ) {
+	if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+		error_log(
+			sprintf(
+				/* translators: Placeholders are numbers, versions of PHP in use on the site, and required by VaultPress. */
+				esc_html__( 'Your version of PHP (%1$s) is lower than the version required by VaultPress (%2$s). Please update PHP to continue enjoying VaultPress.', 'vaultpress' ),
+				esc_html( phpversion() ),
+				VAULTPRESS__MINIMUM_PHP_VERSION
+			)
+		);
+	}
+
+	/**
+	 * Outputs an admin notice for folks running an outdated version of PHP.
+	 *
+	 * @todo: Remove once WP 5.2 is the minimum version.
+	 *
+	 * @since 2.0.0
+	 */
+	function vaultpress_admin_unsupported_php_notice() { ?>
+		<div class="notice notice-error is-dismissible">
+			<p><?php esc_html_e( 'VaultPress requires a more recent version of PHP and has been paused. Please update PHP to continue enjoying VaultPress.', 'vaultpress' ); ?></p>
```

Comment:
I like how the error_log above actually includes version numbers in the message. Can we add those here?

Reply:
Sounds good. Done in 4b59173c469f06ca8177345ee3e3b9be5e17e46d

![image](https://user-images.githubusercontent.com/426388/59356025-5d989600-8d28-11e9-8746-0c3a8366e1db.png)